### ISSUE TITLE
Missed place files in the Software/Arduino directory

### ISSUE THREAD

**Issue by GQ722** [Protocentral/Pulse#3]
The content:
 Pulse-master/Software/Arduino/Protocentral_MAX30100.h	
and
 Pulse-master/Software/Arduino/Protocentral_MAX30100.cpp 
files must be place inside of the 
 Pulse-master/Software/Arduino/example/PPG_plot
directory.

Reason: If you try to compile the PPG_plot.ino file with Arudino IDE, it will not compile, and hence you cannot upload it to the Arduino.

**Comment by mayoralink** [comment]
Arduino compiles fine, the issue happens in Processing, forgot to mention that

**Comment by mayoralink** [comment]
Grab both files and compress them into a .zip file. Afterwards, import them to your Arduino library via Arduino>Sketch>Include Library>Add ZIP Library...>"File location"

**Comment by GQ722** [comment]
Yes, Thank you that now compiles and uploads to the Arduino UNO; however, when I run the processing files from openview_oximeter directory in pulse-master, the example data won't plot [any] PPG signal or the SpO2 measurements. Here is a GIF I made: https://gyazo.com/1ba9047199aa3df43d47d83f7e53e8e3
On the Arduino I have 'L' led on and 'TX' led on but no reading data!

**Comment by VenkateshBhat** [comment]
Devesh,

1)Are you sure about the i2c lines?
2)Is the RED led on the sensor is glowing?
![pulse](https://cloud.githubusercontent.com/assets/6176131/20823778/44975736-b87d-11e6-9adc-e6e212ee6449.jpg)

**Comment by GQ722** [comment]
I'm not using the Pulse from ProtoCentral. I have the following Max 30100
![max30100](https://cloud.githubusercontent.com/assets/6555859/20855561/65c01494-b8cc-11e6-9f55-56e03f67adfe.jpg)

I connect all the wires as stated but there is no red led powering up. What could I possibly be doing wrong?

Also, what is the purpose of the example data in the software/Arduino/example directory?

**Comment by VenkateshBhat** [comment]
The example sketch suppose to plot RED and IR PPG signal.
Could you please contact the Manufacture of the board for further queries.

**Comment by GQ722** [comment]
I read the example sketch file(PPG Plot.ino), it make sense that it suppose to plot RED and IR PPG signal. I open the PPG Plot file, compile, and upload to Arduino. Then I open the Processing GUI program to run the uploaded code from the Arduino. Now I select my port (COM1) then click start.
Well then I should see the output of PPG plot and the data of the example file for SpO2. However, I don't see any output whatsoever and about 10 seconds later my SpO2 count changes to 0.0 from 0.

I don't think I'm not doing anything wrong but what do you thing the problem could be?

**Comment by VenkateshBhat** [comment]
I do not see any issues with the Library/sketchs here. Its pretty much straight forward that I2c lines to be hooked up to the micro-controller.  **RED led on the sensor has to glow** with full intensity if everything is fine.
As of know I'm unsure about the hardware,please do contact the **original manufacture** .

**Comment by GQ722** [comment]
Hey,
We replaced our sensor with another MAX30100 and now it works, which is great! I found out that the SpO2 measurement still need some work. It is very inconsistent, the number fluctuates a whole lot and it is hard to get fixed reading. Overall, the program is very nice and Thank you! You can close the issue.

**Comment by 0xulver** [comment]
@GQ722 I just got the same sensor that you linked. Do you mean that the kind of sensor itself does not work or was it your particular sensor that was broken? What is the sensor you have working now in that case and how did you make it work?

**Comment by GQ722** [comment]
@m-brantheim We bought 2 sensor and they both didn't work at all. We couldn't get a red led in the sensor to light up what-so-ever. So we thought the problem was in the sensor not with our wiring or the software. So we got a new MAX30100 sensor (the following image). 

We hooked up the wires to the arduino, gave a power from USB and it worked. The PWR led (green) turned on and the RED led turned on as well. 

So then we concluded that the other sensor was faulty or weren't designed to worked with this project and we returned it.

In several cases the RED led didn't turn on for some reason in the new sensor as well. We had to upload the Arduino code given in the example folder > PPG_Plot.ino file to Arduino and the RED led seemed to turn on and we were able to get the SpO2 reading. Hope this helps.

![1447175080_heart_ra](https://cloud.githubusercontent.com/assets/6555859/21438216/e3ee26a6-c855-11e6-80ab-309c70d1b3c0.jpg)

**Comment by pulse3** [comment]
@GQ722: I bought the same sensor from mIkroe, but I have been struggling find a way to hook up the sensor to the Arduino board. Can you let me know how you wired the sensor to the board? I am new to this and really need your help. I will appreciate any help, especially if you could include a picture of your wired device.

**Comment by GQ722** [comment]
@pulse3
We followed the exact setup that is mention by the author. It's on the README.md file of this repo.

If you know a little about Electrical Engineering then you should not have any trouble making the above setup. Look at the picture below if you need more help, this is the best picture I have.
![20161212_105432](https://cloud.githubusercontent.com/assets/6555859/22662581/8f38dbd8-ec77-11e6-8d6f-8ef1e95e403f.jpg)

**Comment by pulse3** [comment]
@GQ722 : Thank you very much. I appreciate it
